.circleci/config.yml

```diff
@@ -1,28 +1,71 @@
-version: 2
-jobs:
-  build:
+version: 2.1
+commands:
+  runtests:
+    steps:
+      - checkout
+      - restore_cache: # Download and cache dependencies
+          key: libhoney-java-{{ checksum "pom.xml" }}
```

Comment:
Are the dependencies going to be the same (in terms of content, not just the list) between java versions? This cache does not differentiate java versions, so all versions will be pushing their dependencies in to the same part of the filesystem. Is that OK? Would it work better to have a different cache per version of java? If so, it might fit in well with my suggestion below to use parameters to differentiate versions instead of repeating stanzas.

Reply:
Java/Maven do not specify dependencies based on JRE version. The pom dependencies (e.g.  org.apache.httpcomponents:httpasyncclient:4.1.3) will be the same regardless of whether using JDK 5-12. If a version compatible with language upgrades is needed then the pom version will need to be manually modified which affects all builds.

Nevertheless, since these are building in parallel I have added JDK version to the key to avoid any possible concurrency issues. If there are space/cost issues associated with multiple caches, this change can be reverted.